EIPS/eip-5298.md

```diff
@@ -0,0 +1,71 @@
+---
+eip: 5298
+title: ENS as Token Holder
+description: An interface for holding tokens by ENS.
+author: Zainan Victor Zhou (@xinbenlv)
+discussions-to: https://ethereum-magicians.org/t/erc-eip-5198-ens-as-token-holder/10374
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-07-12
+requires: 137, 721, 1155
+---
+
+## Abstract
+An interface for smart contract to become a holder of tokens by honoring ENS ownership.
+This specification supports [EIP-721](./eip-721.md) and [EIP-1155](./eip-1155.md) tokens and is expandable to future token standard or other standard.
+
+## Motivation
+Currently, if someone wants to receive a token, they have to set up a wallet address. By creating a smart contract that conforms to this interface specification, it is possible to decouple the transfer of ownership from the existence of wallet addresses.
+
+## Specification
+
+1. Any conforming smart contract must conform to `ERC721TokenReceiver` defined in [EIP-721](./eip-721.md) and `ERC1155TokenReceiver` defined in [EIP-1155](./eip-1155.md).
+
+```solidity
+interface IERC_ENS_AS_HOLDER is ERC721Receiver, ERC1155Receiver {
+}
+```
+
+## Rationale
+1. We choose ENS because it's a scoped ownership namespace.
+This ERC shall also be able to work in other scoped ownership namespace approach.
+
+## Backwards Compatibility
+No backward compatibility issues were found.
+
+## Reference Implementation
+```solidity
+contract ENSTokenHolding is ERC721TokenReceiver, Initializable {
+  address ENSRootRegistry _ensRoot;
+  mapping<address /* of ERC721 */ => tokenId=> string /* of owner*/)> _erc721TokenHolders;
```

Comment:
This line isn't valid solidity.

Reply:
None of this code would actually work 😕

Reply:
Right, it is just some peusdo code to help myself think. Still working in progress, let me remove it for now, cause it can be misleading for reader